### PR TITLE
Update android project setup with new all-android recipe

### PR DESCRIPTION
**Pull request by Restioson**
Very basic android dev setup in case things need to be tested in a mobile environment.

For me, usually desktop is sufficient, but since I am on Linux/Windows and have no access to an iOS device or Mac with XCode, Android is the easiest option for mobile debugging.

### CODE REVIEW

**Comment by Restioson** [comment]
PR/issue #1000! :tada:

File: README.md [hunk 1]
```diff
@@ -89,6 +89,16 @@ The following command will build and start all the necessary services, including
 just all-ios
 ```
 
+### Run the mobile-app on the Android simulator
+
+Note: Ensure that the Android simulator is running on your machine so it can be selected as target.
+
```
Comment (da-kami) [inline README.md:95]:
```suggestion
Note: Ensure that you have run `just deps-android` so the necessary targets are added for the build process.

```

**Review by da-kami — COMMENTED** [review]
Compared to the iPhone Simulator the Emulator is... buggy... Somehow the UI keeps reporting that it is unresponsive 😅

I tried to test this on my machine, but was unable to go past the initial Email signup screen. I think somehow the rust backend is not built/started correctly. 

I keep running into this error when I try to submit the email in the app:

```
══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
The following ArgumentError was thrown while handling a gesture:
Invalid argument(s): Failed to load dynamic library 'libnative.so': dlopen failed: library "libnative.so" not found

When the exception was thrown, this was the stack:
#0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11:43)
#1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22:12)
#2      api (package:get_10101/ffi.dart:22:22)
#3      api (package:get_10101/ffi.dart)
#4      _WelcomeScreenState.build.<anonymous closure> (package:get_10101/features/welcome/welcome_screen.dart:78:23)

```

File: justfile [hunk 2]
```diff
@@ -241,6 +241,9 @@ all: services gen native run
 # Run everything at once, tailored for iOS development (rebuilds iOS)
 all-ios: services gen ios run
 
+# Run everything at once, tailored for Android development (rebuilds Android)
+all-android: services gen android run-local-android
```
[the remark this answers is not in the document]
Comment (Restioson) [reply justfile:245]: Can confirm it does work for simulator!

**Comment by Restioson** [comment]
> Is there yet another target missing for running this in the emulator on Mac? Any ideas @Restioson

This branch is missing a change on main which builds libnative.so for x86, so with that integrated it should work!

**Comment by Restioson** [comment]
I've updated this from main, but it should work

**Comment by da-kami** [comment]
Retesting!

**Comment by da-kami** [comment]
> Retesting!

I ran into a separate error:

```
══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
The following ArgumentError was thrown while handling a gesture:
Invalid argument(s): Failed to load dynamic library 'libnative.so': dlopen failed: cannot locate
symbol "__extenddftf2" referenced by
"/data/app/~~R9JQuf4SHIHcuppjYxBypA==/finance.get10101.app.local-X7s4VBga0zoY5UDPecX7kw==/lib/x86_64/libnative.so"...

When the exception was thrown, this was the stack:
#0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11:43)
#1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22:12)
#2      api (package:get_10101/ffi.dart:22:22)
#3      api (package:get_10101/ffi.dart)
#4      _WelcomeScreenState.build.<anonymous closure> (package:get_10101/features/welcome/welcome_screen.dart:78:23)

```

I think this must be related to my setup somehow, but don't have the time to dig into this.

I approved the PR, because I don't think this problem is related to the changes :)

**Review by da-kami — APPROVED** [review]

**Pull request merged**